tests/e2e/lifecycle/mcg/test_mcg_namespace_lifecyle.py

```diff
@@ -0,0 +1,196 @@
+        7. Delete namespace resource and bucket
+
+        """
+        data = "Sample string content to write to a S3 object"
+        object_key = "ObjKey-" + str(uuid.uuid4().hex)
+
+        aws_s3_creds = {
+            'access_key_id': cld_mgr.aws_client.access_key,
+            'access_key': cld_mgr.aws_client.secret_key,
+            'endpoint': constants.MCG_NS_AWS_ENDPOINT,
+            'region': config.ENV_DATA['region']
+        }
+
+        # Noobaa s3 account details
+        user_name = "noobaa-user" + str(uuid.uuid4().hex)
+        email = user_name + "@mail.com"
+
+        # Create the namespace resource and verify health
+        aws_res = ns_resource_factory()
+
+        # Create the namespace bucket on top of the namespace resources
+        ns_bucket = bucket_factory(
+            amount=1, interface='mcg-namespace',
+            write_ns_resource=aws_res[1],
+            read_ns_resources=[aws_res[1]]
+        )[0].name
+        logger.info(f"Namespace bucket: {ns_bucket} created")
+
+        # Noobaa S3 account
+        user = NoobaaAccount(mcg_obj, name=user_name, email=email, buckets=[ns_bucket])
+        logger.info(f"Noobaa account: {user.email_id} with S3 access created")
+
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=user.email_id,
+            actions_list=['PutObject', 'GetObject'],
+            resources_list=[f'{ns_bucket}/{"*"}']
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(f'Creating bucket policy on bucket: {ns_bucket} with wildcard (*) Principal')
+        put_policy = put_bucket_policy(mcg_obj, ns_bucket, bucket_policy)
+        logger.info(f'Put bucket policy response from Admin: {put_policy}')
+
+        # Getting Policy
+        logger.info(f'Getting bucket policy on bucket: {ns_bucket}')
+        get_policy = get_bucket_policy(mcg_obj, ns_bucket)
+        logger.info(f"Got bucket policy: {get_policy['Policy']}")
+
+        # MCG admin writes an object to bucket
+        logger.info(f'Writing object on bucket: {ns_bucket} by admin')
+        assert s3_put_object(mcg_obj, ns_bucket, object_key, data), "Failed: PutObject"
+
+        # Verifying whether Get & Put object is allowed to S3 user
+        logger.info(
+            f'Get object action on namespace bucket: {ns_bucket}'
+            f' with user: {user.email_id}'
+        )
+        assert s3_get_object(user, ns_bucket, object_key), "Failed: GetObject"
+        logger.info(
+            f'Put object action on namespace bucket: {ns_bucket}'
+            f' with user: {user.email_id}'
+        )
+        assert s3_put_object(user, ns_bucket, object_key, data), "Failed: PutObject"
+
+        # Verifying whether Delete object action is denied
+        logger.info(
+            f'Verifying whether user: {user.email_id} '
+            f'is denied to Put object after updating policy'
+        )
+        try:
+            s3_delete_object(user, ns_bucket, object_key)
```

Comment:
If 'AccessDenied' is expected then this won't detect if object is deleted successfully.

Reply:
This is not meant to delete an object, it is to check whether "DELETE" S3 action is denied access as per the policy the set.
I have added an else check to assert if "Delete" action is it not denied access.

Reply:
Nice solution